### DISCUSSION TITLE
How-to Migrate coordinator from ConBee II to SONOFF?

### DISCUSSION THREAD

**Discussion by domlen** [Koenkk/zigbee2mqtt#14574] [Devices]
Hi all,

Currently, I'm using the following setup to run zigbee2mqtt within home assistant: A **ConBee II coordinator** and **two SONOFF routers**

<img width="543" alt="image" src="https://user-images.githubusercontent.com/2361464/197170018-6bf0038f-5f9d-4fe1-8c84-9357245a3ea2.png">

I would like to **migrate the coordinator from ConBee II to another SONOFF stick**, so my question is: what is the best way, the best process, and what are the necessary steps to do so? 

The FAQ ([How do I migrate from one adapter to another?](https://www.zigbee2mqtt.io/guide/faq/#how-do-i-migrate-from-one-adapter-to-another) and [What does and does not require repairing of all devices?](https://www.zigbee2mqtt.io/guide/faq/#what-does-and-does-not-require-repairing-of-all-devices)) does not specifically cover the ConBee II stick and unfortunately I'm not aware of and was not able to find out, which chipset it uses. So I just want to be 100% sure to do the right thing. 

Thank you in advance!

Greetings,

Dominik

**Comment by Koenkk** [comment]
Conbee II -> SONOFF requires repairing.

**Comment by domlen** [comment]
Thanks!

What exactly does repairing mean/contain? Starting from scratch, naming
everything new, adjusting every automation etc. or just the act of
physically repairing (enable join all and pushing buttons)?

**Reply by Koenkk** [reply]
If you are using HA I would recommend the following:
- In the z2m configuration set `homeassistant: false`
- Stop z2m
- Delete the `data/database.db`
- Update the port in the z2m configuration.yaml
- Power down all routers in your network 
- Start z2m and start pairing the devices one by one
- Once done, stop z2m and set `homeassistant: true`
- Start z2m and HA will think nothing has happened (so automations keep working)

**Reply by bavo** [reply]
@Koenkk small follow up question because i will be doing the same in a few days. 

Is it possible to move de the zigbee2mqtt container (HA add-on) to a different desktop/system with the conbee II stick and start a new container in Home assistant with the Sonoff stick?
This would allow me to move the devices one by one from conbee II to sonoff and force the pairing mode for my Hue devices. Because Hue devices automatically start pairing mode when removed. But the process described above does not force pairing mode (i think?). 

I'm having problems with the [reset method](https://www.zigbee2mqtt.io/devices/3216431P6.html#hue-dimmer-switch) using the dimmer switch. Therefore my current process requires me to force the hue bulb back into the hue system and remove it from there. This forces the light into a repairing mode but this is really cumbersome. 

Any idea's or suggestions to make this process easier? 

Thx :heart:

**Reply by Danit2** [reply]
@bavo
Yes you can do it like that.
But i have done it like this:

- Install the Edge AddOn on HomeAssistant.
- Copy the zigbee2mqtt and paste the folder and rename it to zigbee2mqtt2. 
- Then rename in the config of the Edge AddOn the data_path to zigbee2mqtt2.
- Set in the zigbee2mqtt2 config  `homeassistant: false`
- Then change in the normal AddOn the serial path of the coordinator und delete the database file and set `homeassistant: false`
- Then you can start both AddOn's. 

Now you can delet the devices in the Edge version an join in the normal version. 

At the end you can change in the normal config  `homeassistant: false` to  `homeassistant: true` and restart the AddOn. 

For me this wase working fine.

**Comment by linuxstony** [comment]
@Koenkk sorry for the necromancing of this thread - but is the repairing still necessary in the current version of z2m?
Currently my config specifies adapter: auto and the port /dev/ttyACM0. Are the devices stored on the stick itself?

**Reply by manuelkue** [reply]
From my understanding, most Zigbee devices save the coordinator's "unique" `ieeeAddr` when beeing paired.
So, if your coordinator changes - or better the `ieeeAddr` of your coordinator - those devices won't work with this new coordinator anymore. So [the docs](https://www.zigbee2mqtt.io/guide/faq/#how-do-i-migrate-from-one-adapter-to-another) should still be valid.

When you can modify the coordinator's `ieeeAddr` re-pairing might not be needed. See here:
https://www.zigbee2mqtt.io/guide/adapters/flashing/copy_ieeaddr.html 

Unfortunately I don't know if changing the ConBee's `ieeeAddr` is possible. If this is feasible, there is a chance that the devices will work afterwards as if nothing had changed.

**Reply by Koenkk** [reply]
yes repairing is still needed

**Reply by TomKassel** [reply]
I also have a Conbee II and 186 devices. I would like to switch to another coordinator, but repairing is an imposition with the amount of devices. It should be much easier to change the coordinator.

**Reply by Koenkk** [reply]
I agree, but currently we don't have a maintainer for the Conbee adapter  integration (Dresden Elektronik did in the past), so it is unlikely that the backup functionality will be added.

**Reply by TomKassel** [reply]
After a long research, I successfully migrated my Conbee2 to an SLZB-06mg24 today without repairing.

**Reply by diedvdyk** [reply]
Please share how you did it please ?

**Reply by TomKassel** [reply]
I use Z2M and Homeseer.
I have Home Assistant on another system for testing. Unfortunately, I had decided on the Conbee2 at that time.
For a change, it is absolutely necessary that you can flash the IEEE address of the new coordinator! 
This is easily possible with the SMLIGHT 06 adapters via the web interface.

This is how I proceeded: 

In Z2M creating a backup of database.db 

SMLight Adapter Firmware and Software Updated Copied the IEEE address from the Conbee2 and flashed the SM Light Adapter with it 

The software Zigpy-CLI installed on the HA test system https://github.com/tube0013/tubeszb_addons/blob/9aa1046912660c0bbbc9b15928aa663313124ec4/tzb-zigpy-cli-tools/DOCS.md

Created a backup of the Conbee2 and performed a restore to the SMLight adapter 

In Z2M setup the new adapter (configuration.yaml). Start Z2M, there should be no errors. 

However, the devices have disappeared because Z2M creates a new database.db with a new coordinator with a different chipset. 
The new database.db consists of only one line (the coordinator) 
Open the saved databas.db in the editor and read the ID from the first line and thus overwrite the ID in the new database.db. (1st line, coordinator) 
Then just copy from line 2 to the end from the old database.db and attach it to the new database.db and save.

Start Z2M.

**Reply by diedvdyk** [reply]
Thank you very much. 
Finally succeeded to migrate away from my Conbee2 to SkyConnect.

**Comment by haegar33** [comment]
I went thru this process a month ago as my Raspee adapter stopped working in my Pi3.  Having restored all the content of /opt/zigbee2mqtt/data I was pleased to see that with the new Sonoff Z3 Adapter appeared in the front GUI list...
however it turned out they most of the devices were not working until you did  a repairing  :(

**Comment by Testmangh** [comment]
To migrate from SONOFF Dongle P to SLZB06M I need to repair all the devices as well?

**Reply by Testmangh** [reply]
Found a [video ](https://www.youtube.com/watch?v=MJ3UZw-Zkn0)here, whatching it right now.

**Reply by chris-1243** [reply]
Check here

https://github.com/Koenkk/zigbee2mqtt/discussions/26716

**Reply by Testmangh** [reply]
Gonaa try later today GMT-3.

**Comment by Testmangh** [comment]
Worked fine, replace the Sonoff dongle P (zStack) for one SLZB06M (Ember) it took some while for the device start respoding but the eventually did.

Now I have a question, the sonoff dongle P that was replaced can be used on another zigbee2mqtt instance (a test/lab instance) on the same house?

**Reply by chris-1243** [reply]
Yes of course. Do not use the same channel and I would generate new network keys panID and ext-panID to have to completely seperate networks
